Add Content Vertical Spacing control

cristian-ungureanu opened the pull request:
### Summary
- Added a new control for Content Vertical Spacing.
- The control allows you to customize the content's top and bottom margins
Note: I also added support for CPT module, that's treated in a PR that will be made in Neve Por repo

### Will affect the visual aspect of the product
NO

### Screenshots <!-- if applicable -->
https://vertis.d.pr/v/A9cpqv

### Test instructions
- Test the Global Vertical Spacing that applies for both pages and posts
- Save the Global Vertical Spacing, refresh the customizer page and go to Customizer -> Layout -> Single post -> Page settings. Move the control to "Custom" and check if the default value is the same as Global Vertical Spacing
- Test the Post Vertical Spacing
- Test the Page Vertical Spacing

## Check before Pull Request is ready:

* [x] I have [written a test](CONTRIBUTING.md#writing-an-acceptance-test) and included it in this PR
* [x] I have [run all tests](CONTRIBUTING.md#run-tests) and they pass
* [x] The code passes when [running the PHP CodeSniffer](CONTRIBUTING.md#run-php-codesniffer)
* [x] Code meets [WordPress Coding Standards](CONTRIBUTING.md#coding-standards) for PHP, HTML, CSS and JS
* [x] [Security and Sanitization](CONTRIBUTING.md#security-and-sanitization) requirements have been followed
* [x] I have assigned a reviewer or two to review this PR (if you're not sure who to assign, we can do this step for you)

<!-- Issues that this pull request closes. -->
Closes Codeinwp/neve-pro-addon#1802.
<!-- Should look like this: `Closes #1, #2, #3.` . -->

preda-bogdan reviewed (approved):
The code looks good, there are a lot of changes for review so I'll ask for others to also have a look.

mskapusuz reviewed (approved):
Hey Cristi, looks awesome! 🚀 🎉 

I've just left a few suggestion you can evaluate that if you would like. Apart from that, I can suggest;
* We can have a class constant for default control value instead hard-coded `inherit` [ref](https://github.com/Codeinwp/neve/blob/cc1a10efb512a849ab295f12472ace1f71921a42/inc/customizer/options/layout_single_post.php#L252)
* Also, as I see you created some class constants on [here](https://github.com/Codeinwp/neve/blob/cc1a10efb512a849ab295f12472ace1f71921a42/inc/core/settings/config.php#L88-L93) we can make sure they are used all required places and there is no any hard-coded string around.

I hope these are beneficial you!
Thanks!

cristian-ungureanu commented:
Hello @preda-bogdan, I made some changes after Irinel reported some issues. Could you review the changes I made? Specifically, I added the 'dependsOn' parameter and implemented it for the Spacing control. This was necessary because the Spacing control should only be shown if the buttons control is set to "Custom." Previously, I was using "active_callback" to achieve this, but I encountered a problem. When we filtered the result for CPTs, and the value for the CPT was set to "Custom" while the value for Posts was "Inherit," the spacing control wasn't being hidden because its value was being filtered by the CPT's control.

mskapusuz reviewed (commented):
Looks good 🚀  I've left a few comments.

Apart from that, would be great if we can reduce the strings which needs localization.
